src/client/Microsoft.Identity.Client/Internal/Requests/Silent/CacheSilentStrategy.cs

```diff
@@ -152,16 +153,17 @@ private async Task<AuthenticationResult> RefreshRtOrFailAsync(CancellationToken
         private async Task<AuthenticationResult> CreateAuthenticationResultAsync(MsalAccessTokenCacheItem cachedAccessTokenItem)
         {
             var msalIdTokenItem = await CacheManager.GetIdTokenCacheItemAsync(cachedAccessTokenItem).ConfigureAwait(false);
-            var tenantProfiles = await CacheManager.GetTenantProfilesAsync(cachedAccessTokenItem.HomeAccountId).ConfigureAwait(false);
+            var account = await CacheManager.GetAccountAssociatedWithAccessTokenAsync(cachedAccessTokenItem).ConfigureAwait(false);
```

Comment:
I'm wondering if we can go 2 steps further with this:

1. Eliminate `GetTenantProfilesAsync` completely. It does not seem to be needed, you only use it `AcquireTokenOnBehalfOf`, which can also get an Account object instead. `GetTenantProfilesAsync` can be made private.

2. Merge `GetIdTokenCacheItemAsync` with `GetAccountAssociatedWith...` - since we always make these 2 calls one after the other. There is no code path that gets just the IdToken but not the account. 

Thoughts?

Reply:
I think it makes sense to make `GetTenantProfilesAsync` private and add to `GetAccountAssociatedWithCache`. But leave `GetIdTokenCacheItemAsync` separate since the `TenantProfiles` needs all the id tokens for the `HomeAccountId` and IdToken is for the particular key with partitioning. Combining Tenant profiles and Id token will not have any impact on the cache.